### PR TITLE
docs-fix: replace the links of wikipedia by the links of the famous textbook

### PR DESCRIPTION
**Pull request by Mingbo-Lee**
Usually, people don't use Wikipedia as a reference for machine learning algorithms, so I replaced the Wikipedia links with links  of  classic textbook  Pattern Recognition and Machine Learning.

### CODE REVIEW

**Review by warriorpaw — CHANGES_REQUESTED** [review]
First of all, here is a brief document for LR component, not an academic paper. Wikipedia is concise and accurate enough for users who do not known LR to quickly understand what they see.

Secondly, use a textbook as reference is too heavy for a brief document. Developers don’t need to read it, and beginners who are not familiar with it can’t understand it.

If there is a better concise document for LR than Wikipedia, please submit it in the follow-up commit, thanks.

**Comment by Mingbo-Lee** [comment]
Hi, @warriorpaw Thank you for you suggestion very much.
I try to find a better concise document for LR than Wikipedia and I found maybe we could supply more help link to help developers and researchers to understand LR quickly instead of  just one link. 
The three different types of link targeting are as follows：
- Wikipedia：Easy to understand and concise;
- implementation in scikit-learn:  It is more concise and provides a traditional implementation for users to compare with secretflow.
- Textbook reference: comprehensive and academic.


In this case, the document of the SecretFlow is more comprehensive and We give our readers more choice and give them the choice.
Besides, the LR description in the book Dive into Deep Learning is worthy considering: https://d2l.ai/chapter_linear-regression/linear-regression.html#basics, which is concise and comprehensive.
However, the [Logistic regression](https://d2l.ai/chapter_linear-classification/softmax-regression.html#linear-model) in this book maybe does not meet our needs so that I did't commit, could you comment about  the situation of the book.

**Comment by warriorpaw** [comment]
> Hi, @warriorpaw Thank you for you suggestion very much. I try to find a better concise document for LR than Wikipedia and I found maybe we could supply more help link to help developers and researchers to understand LR quickly instead of just one link. The three different types of link targeting are as follows：
> 
> * Wikipedia：Easy to understand and concise;
> * implementation in scikit-learn:  It is more concise and provides a traditional implementation for users to compare with secretflow.
> * Textbook reference: comprehensive and academic.
> 
> In this case, the document of the SecretFlow is more comprehensive and We give our readers more choice and give them the choice. Besides, the LR description in the book Dive into Deep Learning is worthy considering: https://d2l.ai/chapter_linear-regression/linear-regression.html#basics, which is concise and comprehensive. However, the [Logistic regression](https://d2l.ai/chapter_linear-classification/softmax-regression.html#linear-model) in this book maybe does not meet our needs so that I did't commit, could you comment about the situation of the book.


Just keep one, the documentation provided by d2l.ai is quite good, thanks very much for your helping to find the documentation.

And pls pay attention to the failure of ci.

**Comment by Mingbo-Lee** [comment]
Thank you for your suggestion! @warriorpaw I replaced the  links of LR with the links from [the d2l.ai](https://d2l.ai/). I also replaced  the link of minibatch SGD with the link from [the d2l.ai](https://d2l.ai/).
I read the code and found ``batch_size = 32 ``,usually, the size of dataset could not be 32,  so I think it can  be a mini batch from the dataset. even the title of the [old link](https://stats.stackexchange.com/questions/488017/understanding-mini-batch-gradient-descent)  is understanding-mini-batch-gradient-descent and the statement of the document  may be wrong,the chinese translation of minibatch SGD refer to '小批量梯度下降方法' instead of  '批量梯度下降方法' and then I fixed the bug in the document.
the difference of SGD, batch SGD and mini-batch SGD can refer:
https://datascience.stackexchange.com/questions/53870/how-do-gd-batch-gd-sgd-and-mini-batch-sgd-differ
https://machinelearningmastery.com/gentle-introduction-mini-batch-gradient-descent-configure-batch-size/

**Comment by Mingbo-Lee** [comment]
I don't know the reason why the ci fail, maybe the Chinese documents did't  not correspond to English documents? Could you help me please?

**Review by warriorpaw — APPROVED** [review]
LGTM
